### DISCUSSION TITLE
Microsoft did it again ... 'sudo'

### DISCUSSION THREAD

**Discussion by mwallner** [gerardog/gsudo#331] [Q&A]
So MSFT decided to make it's own version of sudo?

=> https://github.com/microsoft/sudo

is this going to change anything? 
for me I currently see no reason to use 'sudo.exe' over 'gsudo.exe' (that said, always using the sudo alias) - especially because I'm mostly using it on Win10 LTSC and Server OSes - which will most likely never be supported by msft's sudo.

**Comment by Enivex** [comment]
They specifically mention gsudo in their blog post https://devblogs.microsoft.com/commandline/introducing-sudo-for-windows/

I don't see an issue here.

**Comment by septsea** [comment]
Note that Microsoft's `sudo` precedes the alias (i.e. `sudo`) of `gsudo`.

**Reply by septsea** [reply]
![image](https://github.com/gerardog/gsudo/assets/57337516/95ab7cd5-3073-44b2-a6ee-15df5d990e01)

**Reply by septsea** [reply]
I have been used to gerardog's `sudo`; the new `sudo` does not work very well.

**Reply by e-t-l** [reply]
I'm curious, why do you say Microsoft's `sudo` doesn't work well? I'm inclined to agree, at least I like using gerardog's version better (the red hashtag is a nice easy visual reminder of the elevation, for one thing.) But I'm curious if there's something specific you found that doesn't work.

**Reply by gerardog** [reply]
If you want an updated comparison, check [here](https://gerardog.github.io/gsudo/docs/gsudo-vs-sudo). However, keep in mind that at the time that comment was posted, ms-sudo had several issues, including problems with input redirection, not returning the exit code, and even [poor security reviews](https://www.tiraniddo.dev/2024/02/sudo-on-windows-quick-rundown.html).  

Most of these issues have been fixed now, but *_the one major ms-sudo issue that truly scares me_* is this:  

⚠️If you run `sudo del *.*`, it will elevate to delete files in the current folder—*unless* the user has `new-window` set as the default elevation mode. In that case, it will execute `del *.*` in `C:\windows\system32\`, potentially destroying your OS. 

![image](https://github.com/user-attachments/assets/e8cd4b74-8262-4d5e-b1fe-0f041589288c)

I think this is not just a bug— it’s a catastrophic failure with far-reaching consequences. I reported it eight months ago... They still shipped Windows 11 24H2 with this issue, which reflects poorly on the project’s governance and priorities.

The worst part is that their failure to address this issue highlights the political dynamics within that project. There is a non-zero probability that it will end up abandoned and crippled, much like `runas.exe`.  

Aside from that, there are countless reasons why gsudo outperforms ms-sudo in every way—from user experience (e.g., no pedantic case-sensitive arguments) to features like PowerShell elevation, PowerShell auto-complete, a credentials cache, and better integration with other shells like Bash, WSL, etc.

**Reply by septsea** [reply]
@e-t-l,

Hereinafter, gsudo is the sudo equivalent for Windows made by gerardog, and ms-sudo is Microsoft's work.

- I have been on the Canary channel (of Windows Insider Programme) for many years (including the old Dev channel). When ms-sudo was available on my PC, I was excited to enable it. However, I had some problems in enabling it (in the "For developers" page), and I resorted to [some tricks](https://github.com/microsoft/sudo/issues/25).

- ms-sudo did not have `sudo !!` (or its equivalents).

- ms-sudo did not have the "credentials cache" (or its equivalents).

Because I found that ms-sudo did not work up to my expectation, I disabled it. I did not find the major issues that gerardog just mentioned.

**Reply by gamingdivy** [reply]
that's why, microsoft's SUDO IS LITERALLY TRASH THEY SHOULD REMOVE IT OR YALL SHOULD REMOVE IT

**Comment by gamingdivy** [comment]
in Windows 10, nothing changes, for Windows 11 users, IT CHANGES. and I LITERALLY HATE MICROSOFT'S SUDO AS MUCH AS YOU GUYS DO.

**Comment by DataComputerSecurity** [comment]
**How is it possible to have sudo on Windows? Isn't sudo a command from Linux distributions? It's actually pretty unique—I never expected an app like this to exist. 👍**
